Adding compatibillity to older otb version

danilopucci opened the pull request:
Theese commits adds compatibillity to older otb versions, for who wants to edit files from 7.4-7.92 servers.
I tried to follow all the coding patterns of project and I named the plugin as PluginZero, as One is from 8.0+ and so on.

Fixes issues #4 and #5 (seems to be duplicated)

danilopucci committed:
- Adding files related to PluginZero solution

files are

danilopucci committed:
Changes related to OTBVersion on write/save

- Changed OtbWriter.cs vi.BuildNumber now increments
- OTB Major version are saved according to clients.xml in Remeres Map Editor. Version = 1 to version <=792, version = 2 to 800 to 811 and version = 3 on later versions 
-Few changes in csprojs related to framework version

danilopucci committed:
Solving PluginZero project configurations and adding PluginZero.xml

danilopucci committed:
Update README.md

editting download link on README.md

p0t1nh0 commented:
Do u have a compiled version who works in 7.6 or 7.72?

danilopucci commented:
> Do u have a compiled version who works in 7.6 or 7.72?

Yes! It is on my fork. Readme section "download"

danilopucci commented:
@Mignari Do you have any intention to merge it to master?

WooXZerah commented:
This doesn't support 7.4 even though it does support 7.7.

danilopucci commented:
> This doesn't support 7.4 even though it does support 7.7.

Did you find any incompatibility in my changes? 
Can you give any details about it?
Thanks

WooXZerah commented:
> Did you find any incompatibility in my changes?
> Can you give any details about it?
> Thanks

I just tested it and it didn't work for 7.4 version which uses OTB version 1, your plugin seems to support only version 3 which started on version 7.6, items flags seems to be a bit different on versions prior to this, such as 7.4.

danilopucci commented:
> > Did you find any incompatibility in my changes?
> > Can you give any details about it?
> > Thanks
> 
> I just tested it and it didn't work for 7.4 version which uses OTB version 1, your plugin seems to support only version 3 which started on version 7.6, items flags seems to be a bit different on versions prior to this, such as 7.4.

Can you provide me the otb that you tested? I will check it and update as soon as possible

WooXZerah commented:
> Can you provide me the otb that you tested? I will check it and update as soon as possible

Sure, here it is.
[items.zip](https://github.com/ottools/ItemEditor/files/5947204/items.zip)

danilopucci committed:
Update PluginZero.xml

Adding 7.x client, otb versions and spr, dat signatures

WooXZerah commented:
@danilopucci these might help with old known byte flags.

https://github.com/opentibia/item-editor/blob/06580b4caacfe4ca1cb59eaafea531acf8988fac/client/item_type.cpp
https://github.com/opentibia/item-editor/commit/59028ea4b8bea5bf154473cdecb1d4a2d02ca481

danilopucci commented:
> @danilopucci these might help with old known byte flags.
> 
> https://github.com/opentibia/item-editor/blob/06580b4caacfe4ca1cb59eaafea531acf8988fac/client/item_type.cpp
> [opentibia/item-editor@59028ea](https://github.com/opentibia/item-editor/commit/59028ea4b8bea5bf154473cdecb1d4a2d02ca481)

It will help for sure! I was looking on RME and TFS (items.cpp) to get the definitions and the flags, going to check your recomendations now 
Thank you

danilopucci committed:
- Reverting some project conf files, which caused conflicts to main repo

danilopucci committed:
- Updating PluginZero proj file to the same framework version as others

danilopucci committed:
Adding checkages of client version while loading .dat

The flags of .dat files on versions 7.x are bit different on their values. Using OTClient sources as a reference, I added the checkages on versions to load that correctly.

danilopucci commented:
@WooXZerah Now things should work correctly. 
It looks that tibia.dat files on versions 7.x are very different, specially on version 7.4. On version 7.55 above things are more similar from each other but still has some differences.
I took as code reference [this piece of OTClient code](https://github.com/edubart/otclient/blob/master/src/client/thingtype.cpp#L134) that consider and adjust the byte flags according to Tibia version properly. The reference that you provided me helped alot! Reading there I could track the differences of the byte flags and I remmembered that OTClient reads .dat files too, so I took a look there and found the code considerations of many versions to retro-compatibillity. Both references are equal on this point.
I tested it using Tibia.dat from Tibia7.4 client and the .otb that you provided me and it works fine. 
Could you please test it?
Download [here](https://github.com/danilopucci/ItemEditor/releases/download/v0.4.3/ItemEditor_v0431.zip) or by going on releases on my fork. ItemEditor_v0431.zip

WooXZerah commented:
@danilopucci Mission accomplised! I tested and it worked perfectly, I'm glad I could help.

@Mignari This deserve some attention, it should be merged to master.

danilopucci committed:
Update MainForm.cs

Changes in logic to find a compatible client. Previously there was only one OTB version to each Tibia version, with the new 7.x support there is more than one Tibia version to the same OTB version (710->730; 740->750 and 790 -> 792).
With theese changes, now the logics consider a list instead of a single client, it reads the list and try to match it by the dat and spr signature

danilopucci committed:
Update PluginZero.xml

Fixing typo in version 780

Feliipe93 commented:
what happened with this ?

danilopucci commented:
> what happened with this ?

Hello Felipe, 

For now, you can use the compiled version on my branch. 

Please, let me know if you find any bug.

rbgadotti commented:
I can't download it,
the link on readme is broken.
Could you help me with that? Really need edit 7.72 items

danilopucci commented:
Try from [releases](https://github.com/danilopucci/ItemEditor/releases)

EDIT: I fixed the download link from my readme. Thanks

danilopucci committed:
Update README.md

mano368 commented:
> Try from [releases](https://github.com/danilopucci/ItemEditor/releases)
> 
> EDIT: I fixed the download link from my readme. Thanks

hey danilopucci, do you know how to solve this problem on windows 11?

![Sem título](https://github.com/ottools/ItemEditor/assets/53128315/9f19a8cd-eb87-4d35-95c5-3d0e7396eadd)

danilopucci commented:
Hey, I never tried to run it on win11. 

Did you tried to compile it? I think if you try to recompile it, the missing libraries or other kind of error will throw and we can investigate it. Or maybe just by recompiling it will fix the problem.

(I do not have win11 installed)

danilopucci committed:
- improve compatibility with older versions

danilopucci committed:
- move 7.8+ to Plugin One

danilopucci committed:
- remove empty line

danilopucci committed:
- bump version 0.4.4

MightyGoober commented:
@danilopucci I'm looking to use your work on a 7.6 server (Goobers).
Do you happen to have a recent copy of the plug files I need to use?

MightyGoober commented:
@danilopucci I found your recompiled download. It works!
AMAZING!

I was actually hoping for an item editor to have an xml or sql export. Mainly so I can store all my items.otb information in tables to be shown on the website.

Any thoughts on this?

danilopucci commented:
Hey @MightyGoob,

I am not sure if I get it correctly, but maybe you would need something on your gameserver to update the database table with your items.otb. This way you will be 100% sure that your lastest version of items.otb will be reflected on your database table, with no dependencies on any other kind of file
